Bal 4425 bullmq refactor

OmerHarell opened the pull request:
* Make queue service abstract
* One place for monitoring 
* Inject redis

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Introduced Redis integration for improved queue management and monitoring capabilities.
  - Added a unified interface for queue operations, streamlining job scheduling and worker registration.

- **Refactor**
  - Simplified queue and worker setup, reducing dependencies and improving error handling.
  - Updated job retry and backoff configurations for more efficient processing.
  - Streamlined module structure and removed unused middleware and monitoring integrations.
  - Enhanced error logging and handling in job processing workflows.
  - Consolidated queue creation and registration with monitoring and UI services.
  - Refined webhook queue handling with improved job processing and error management.
  - Adjusted feature flag checks to ensure explicit control over webhook queue system usage.

- **Chores**
  - Changed environment variable for worker port configuration.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

OmerHarell committed:
refactor(workflows-service): remove unnecessary middleware and update port config

- Remove MetricsAuthMiddleware configuration from WorkerAppModule
- Change retrieved port from 'PORT' to 'WORKER_PORT' in main.worker.ts
- Update BullBoardAuthMiddleware usage in WebhooksModule configuration

OmerHarell committed:
fix(queue): refactor redis connection handling in QueueService

- Remove Redis connection initialization from constructor
- Inject Redis client through dependency injection
- Improve code structure by eliminating unnecessary methods

OmerHarell committed:
refactor(queue): streamline queue registration process and improve imports

- Remove redundant queue registration lines in AlertQueueService
- Integrate BullBoard instance creation within QueueModule
- Simplify dependency handling in WebhooksService and related modules

OmerHarell committed:
refactor(queue): improve alert queue service and worker registration

- Simplify dependency injection for services in AlertQueueService
- Replace queue setup logic with createQueue method and refactor worker processing
- Eliminate unused code and streamline error handling in webhook job processing

OmerHarell committed:
feat(queue): implement job scheduler in alert queue service

- Add job scheduling setup for alert-check jobs with specific intervals
- Update relevant imports and refactor the QueueService to utilize RedisService
- Enhance IQueueService interface with additional documentation and job-related methods

OmerHarell committed:
refactor(queue): simplify queue configuration and job setup

- Remove unnecessary injection of BullBoard instance in AlertQueueService
- Update job options to reduce default attempts for better performance
- Enhance queue service methods for clearer job scheduling and worker registration

jonathan-shemer commented on services/workflows-service/src/common/queue/types.ts:
```diff
@@ -7,3 +7,41 @@ export interface BullBoardInjectedInstance {
+  ): Promise<any>;
+}
+
+export interface QueueOptions<T = any> {
```
why `<T = any>`?

jonathan-shemer commented on services/workflows-service/src/common/queue/types.ts:
```diff
@@ -7,3 +7,41 @@ export interface BullBoardInjectedInstance {
+    jobOpts: {
+      name: string;
+      data: T;
+      opts?: any;
```
why isn't this `JobOptions`

jonathan-shemer commented on services/workflows-service/src/common/queue/types.ts:
```diff
@@ -7,3 +7,41 @@ export interface BullBoardInjectedInstance {
+export interface QueueOptions<T = any> {
+  name: string;
+  concurrency?: number;
+  jobOptions?: {
```
should be `JobOptions`

and
```ts
const interface JobOptions {
    attempts?: number;
    backoff?: {
      type: 'exponential' | 'fixed';
      delay: number;
    };
    removeOnComplete?: boolean | number | { count: number; age: number };
    removeOnFail?: boolean | number | { count: number; age: number };
    priority?: number;
  };
```

jonathan-shemer commented on services/workflows-service/src/alert/alert-queue.service.ts:
```diff
@@ -21,53 +16,37 @@ export class AlertQueueService implements OnModuleInit {
-            delay: 10_000,
-          },
+          attempts: 3,
+          backoff: { type: 'exponential', delay: 10000 },
```
see comments on typing here, below

OmerHarell committed:
fix(webhook): correct forceDirect logic in webhook callers

- Update forceDirect assignment to explicitly check for enabled state
- Ensure consistent behavior for document and workflow state changes

OmerHarell committed:
add feautres

OmerHarell committed:
refactor(queue): streamline job options and improve queue management

- Remove deprecated job option settings for consistency
- Introduce default job options in queue creation function
- Add validation to ensure valid queue names are provided
